### PR TITLE
Port all Python examples to new APIs

### PR DESCRIPTION
**Pull request by teh-cmc**
- [x] All examples run and work as expected
- [x] `rg -c 'rr\.log_' == 0`
- [x] All annoyances are either fixed or have an issue opened

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3545) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3545)
- [Docs preview](https://rerun.io/preview/5748f64bbc3b4a69f872ba83e670c650c30de977/docs) 
- [Examples preview](https://rerun.io/preview/5748f64bbc3b4a69f872ba83e670c650c30de977/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: examples/python/ros_node/main.py
Comment (teh-cmc) [inline examples/python/ros_node/main.py:1]: I cannot run this one!

File: examples/python/live_depth_sensor/main.py
Comment (teh-cmc) [inline examples/python/live_depth_sensor/main.py:1]: I cannot run this one!

File: examples/python/signed_distance_fields/main.py [hunk 1]
```diff
@@ -101,35 +88,40 @@ def log_mesh(path: Path, mesh: Trimesh) -> None:
...
     scale = bs2.scale / bs1.scale
     center = bs2.center - bs1.center * scale
-    rr.log_mesh_file(
-        "world/mesh",
-        mesh_format,
-        mesh_path=path,
-        transform=np.array([[scale, 0, 0, center[0]], [0, scale, 0, center[1]], [0, 0, scale, center[2]]]),
-    )
+
+    mesh3d = rr.Asset3D.from_file(path=str(path))
```
Comment (teh-cmc) [inline examples/python/signed_distance_fields/main.py:95]: Not how I have to cast to `str` :grimacing:

Comment (teh-cmc) [reply examples/python/signed_distance_fields/main.py:95]: Also note how everything is both awful and inconsistent here.

This shouldn't be `Asset3D.from_file`, this should be extra args to `Asset3D.__init__`.

File: examples/python/structure_from_motion/main.py [hunk 2]
```diff
@@ -110,7 +110,7 @@ def read_and_log_sparse_reconstruction(dataset_path: Path, filter_output: bool,
         points3D = {id: point for id, point in points3D.items() if point.rgb.any() and len(point.image_ids) > 4}
 
     rr.log("description", rr.TextDocument(DESCRIPTION, media_type=rr.MediaType.MARKDOWN), timeless=True)
-    rr.log_view_coordinates("/", up="-Y", timeless=True)
+    rr.log("/", rr.ViewCoordinates.RIGHT_HAND_Y_DOWN, timeless=True)
```
Comment (teh-cmc) [inline examples/python/structure_from_motion/main.py:113]: Somehow, for some reason I don't understand, in this specific case logging `ViewCoordinates` at `/` works, despite #3538

File: examples/python/text_logging/main.py
Comment (teh-cmc) [inline examples/python/text_logging/main.py:1]: I just burnt it, it's redundant at this point.

File: examples/python/arkit_scenes/main.py [hunk 3]
```diff
@@ -351,17 +354,19 @@ def log_arkit(recording_path: Path, include_highres: bool) -> None:
         timestamp = f"{round(float(timestamp), 3):.3f}"
         camera_from_world_dict[timestamp] = camera_from_world
 
-    rr.log_view_coordinates("world", up="+Z", right_handed=True, timeless=True)
+    rr.log("world", rr.ViewCoordinates.RIGHT_HAND_Z_UP, timeless=True)
```
[the remark this answers is not in the document]
Comment (teh-cmc) [reply examples/python/arkit_scenes/main.py:357]: https://github.com/rerun-io/rerun/issues/3552

File: examples/python/detect_and_track_objects/main.py [hunk 4]
```diff
@@ -314,7 +320,7 @@ def track_objects(video_path: str) -> None:
     class_descriptions = [
         rr.AnnotationInfo(id=cat["id"], color=cat["color"], label=cat["name"]) for cat in coco_categories
     ]
-    rr.log_annotation_context("/", class_descriptions, timeless=True)
+    rr.log("/", rr.AnnotationContext(class_descriptions), timeless=True)
```
[the remark this answers is not in the document]
Comment (teh-cmc) [reply examples/python/detect_and_track_objects/main.py:323]: https://github.com/rerun-io/rerun/issues/3552

File: examples/python/human_pose_tracking/main.py [hunk 5]
```diff
@@ -25,41 +25,56 @@
...
 
             results = pose.process(rgb)
             h, w, _ = rgb.shape
             landmark_positions_2d = read_landmark_positions_2d(results, w, h)
             if landmark_positions_2d is not None:
-                rr.log_points("video/pose/points", landmark_positions_2d, keypoint_ids=mp_pose.PoseLandmark)
+                rr.log(
+                    "video/pose/points",
+                    rr.Points2D(landmark_positions_2d, class_ids=0, keypoint_ids=mp_pose.PoseLandmark),
+                )
```
[the remark this answers is not in the document]
Comment (teh-cmc) [reply examples/python/human_pose_tracking/main.py:66]: cc @Wumpf, that seems somewhat related to something you fixed recently (I don't remember what exactly)

File: examples/python/human_pose_tracking/main.py [hunk 6]
```diff
@@ -25,41 +25,56 @@
...
-                rr.log_points("person/pose/points", landmark_positions_3d, keypoint_ids=mp_pose.PoseLandmark)
+                rr.log(
+                    "person/pose/points",
+                    rr.Points3D(landmark_positions_3d, class_ids=0, keypoint_ids=mp_pose.PoseLandmark),
+                )
 
             segmentation_mask = results.segmentation_mask
             if segmentation_mask is not None:
-                rr.log_segmentation_image("video/mask", segmentation_mask)
+                rr.log("video/mask", rr.SegmentationImage(segmentation_mask.astype(np.uint8)))
```
[the remark this answers is not in the document]
Comment (teh-cmc) [reply examples/python/human_pose_tracking/main.py:77]: https://github.com/rerun-io/rerun/issues/3553

File: examples/python/ros_node/main.py
Comment (teh-cmc) [reply examples/python/ros_node/main.py:1]: cc @jleibs

File: examples/python/live_depth_sensor/main.py
Comment (teh-cmc) [reply examples/python/live_depth_sensor/main.py:1]: cc @milk

File: examples/python/signed_distance_fields/main.py (hunk 1 above)
Comment (teh-cmc) [reply examples/python/signed_distance_fields/main.py:95]: https://github.com/rerun-io/rerun/issues/3555

File: examples/python/multithreading/main.py [hunk 7]
```diff
@@ -16,7 +16,7 @@ def rect_logger(path: str, color: npt.NDArray[np.float32]) -> None:
         rects_xy = np.random.rand(5, 2) * 1024
         rects_wh = np.random.rand(5, 2) * (1024 - rects_xy + 1)
         rects = np.hstack((rects_xy, rects_wh))
-        rr.log_rects(path, rects, colors=color, rect_format=rr.RectFormat.XYWH)
+        rr.log(path, rr.Boxes2D(array=rects, colors=color))
```
[the remark this answers is not in the document]
Comment (emilk) [reply examples/python/multithreading/main.py:19]: yeah, we shouldn't have a default here methinks

File: examples/python/car/main.py [hunk 8]
```diff
@@ -17,37 +17,38 @@ def log_car_data() -> None:
...
         rr.set_time_sequence("frame_nr", sample.frame_idx)
 
         # Log the camera pose:
-        rr.log_transform3d(
+        rr.log(
             "world/camera",
             rr.TranslationRotationScale3D(sample.camera.position, rr.Quaternion(xyzw=sample.camera.rotation_q)),
         )
-        rr.log_view_coordinates("world/camera", xyz="RDF")  # X=Right, Y=Down, Z=Forward
+        rr.log("world/camera", rr.ViewCoordinates.RDF, timeless=True)  # X=Right, Y=Down, Z=Forward
```
Comment (emilk) [inline examples/python/car/main.py:32]: this should be logged as the Pinhole `camera_xyz` instead

File: examples/python/live_depth_sensor/main.py
Comment (emilk) [reply examples/python/live_depth_sensor/main.py:1]: I don't have a depth camera, and neither does @jleibs I think

**Pull request merged**